docs/side_quests/working_with_files.md

```diff
@@ -0,0 +1,898 @@
+
+data/sampleA_rep1_normal_R1_001.fastq.gz
+```
+
+---
+
+## 1. Basic File Operations
+
+### 1.1 Creating File Objects
+
+Let's start by understanding how to create file objects in Nextflow. In our workflow, we have a string path `data/sampleA_rep1_normal_R1_001.fastq.gz`. This is just a plain string - Nextflow doesn't automatically recognize it as representing a file. To work with files properly in Nextflow, we need to convert string paths into proper file objects using the `file()` method, which provides access to file properties and operations.
+
+Edit the `main.nf` file to include the following:
+
+_Before_:
+
+```groovy title="main.nf" linenums="2"
+// Create a file object from a string path
+myFile = 'data/sampleA_rep1_normal_R1_001.fastq.gz'
+println "${myFile}"
+```
+
+_After_:
+
+```groovy title="main.nf" linenums="2"
+// Create a file object from a string path
+myFile = file('data/sampleA_rep1_normal_R1_001.fastq.gz')
+println "${myFile}"
+```
+
+```bash
+nextflow run main.nf
+```
+
+```console title="File object output"
+ N E X T F L O W   ~  version 24.10.4
+
+Launching `main.nf` [kickass_coulomb] DSL2 - revision: 5af44b1b59
+
+/workspaces/training/side-quests/working_with_files/data/sampleA_rep1_normal_R1_001.fastq.gz
+```
+
+You should only see a single difference, the file path will now be absolute. Note that the full path will change based on where you are doing this training, but it is likely to be something like `/workspaces/training/side-quests/working_with_files/data/sampleA_rep1_normal_R1_001.fastq.gz`.
+
+### 1.2 File Attributes
+
+Why is this helpful? Well now Nextflow understands that `myFile` is a file object and not just a string, we can access the various attributes of the file object.
+
+Let's update our workflow to print out the file attributes:
+
+_Before_:
+
+```groovy title="main.nf" linenums="2"
+// Create a file object from a string path
+myFile = file('data/sampleA_rep1_normal_R1_001.fastq.gz')
+println "${myFile}"
+```
+
+_After_:
+
+```groovy title="main.nf" linenums="2"
+// Create a file object from a string path
+myFile = file('data/sampleA_rep1_normal_R1_001.fastq.gz')
+
+// Print file attributes
+println "File name: ${myFile.name}"
+println "Simple name: ${myFile.simpleName}"
+println "Extension: ${myFile.extension}"
+println "Parent directory: ${myFile.parent}"
+```
+
+Run the workflow:
+
+```bash
+nextflow run main.nf
+```
+
+You'll see various file attributes printed to the console:
+
+```console title="File Attributes Output"
+ N E X T F L O W   ~  version 24.10.4
+
+Launching `main.nf` [ecstatic_ampere] DSL2 - revision: f3fa3dcb48
+
+File name: sampleA_rep1_normal_R1_001.fastq.gz
+Simple name: sampleA_rep1_normal_R1_001
+Extension: gz
+Parent directory: /workspaces/training/side-quests/working_with_files/data
+```
+
+### Takeaway
+
+- The `file()` method creates a Path object from a string
+- You can access file metadata like name, extension, and path
```

Comment:
Should we link out to the docs for a complete list of file attributes that are available?

Reply:
metadata ?
I'd say properties